std/array.d

```diff
@@ -2517,8 +2517,15 @@ Implements an output range that appends data to an array. This is
 recommended over $(D a ~= data) when appending many elements because it is more
 efficient.
  */
-struct Appender(A)
-if (isDynamicArray!A)
+alias Appender(A) = Buffer!(A, false);
```

Comment:
`Appender` with `popFrontN` requires a new name, and I think that `Buffer` is a better name than `PopFrontAppender`. Later on `Appender` can deprecated, or we can keep it forever as alias.
A new optional enum parameter for the constructor is a good idea... I'll prepare a patch for it.